src/routes/users/updateUser.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  RawReplyDefaultExpression,
+  RawRequestDefaultExpression,
+  RawServerDefault,
+  RouteHandler,
+  RouteOptions,
+} from "fastify";
+import UserSchema from "../../schemas/User.json";
+import { UserRequestParams } from "../../types/UserRequestParams";
+import UserRequestParamsSchema from "../../schemas/UserRequestParams.json";
+import { User } from "../../types/User";
+import UserModel from "../../models/user";
+import { ErrorSchema } from "../../types/ErrorSchema";
+import ErrorSchemaJson from "../../schemas/ErrorSchema.json";
+
+type Reply = User | { error: ErrorSchema };
+type UpdateUserRoute = {
+  Body: User;
+  Params: UserRequestParams;
+  Reply: Reply;
+};
+
+const url = "/user/:id";
+
+const handler: RouteHandler<UpdateUserRoute> = async (req, reply) => {
+  try {
+    const { id: userId } = req.params;
+    const { firstName, areaId, lastName } = req.body;
+
+    const newUserFirstName =
+      firstName.charAt(0).toUpperCase() + firstName.slice(1);
```

Comment:
In the current code, when the first name is in uppercase (e.g., LEANDRO) or if characters are presented in a combined format (e.g., AleXANdeR), it does not change to title case properly. I recommend implementing the suggested solution to ensure consistent capitalization regardless of the input format.
```suggestion
  const newUserFirstName = firstName.toLowerCase().replace(/^\w/, (c) => c.toUpperCase());
```

Reply:
@AlvisonHunterArnuero, we will create a helper function for this! @leandro0day is aware of it. Thanks for this excellent suggestion as usual. 😃

Reply:
Sounds good, pal!